edx_arch_experiments/datadog_monitoring/scripts/datadog_search.py

```diff
@@ -0,0 +1,186 @@
+"""
+This script takes a regex to search through the Datadog monitors
+and dashboards.`
+
+For help::
+
+    python edx_arch_experiments/datadog_monitoring/scripts/datadog_search.py --help
+
+"""
+import datetime
+import re
+import types
+
+import click
+from datadog_api_client import ApiClient, Configuration
+from datadog_api_client.v1.api.dashboards_api import DashboardsApi
+from datadog_api_client.v1.api.monitors_api import MonitorsApi
+
+
+@click.command()
+@click.option(
+    '--regex',
+    required=True,
+    help="The regex to use to search in monitors and dashboards.",
+)
+def main(regex):
+    """
+    Search Datadog monitors and dashboards using regex.
+
+    Example usage:
+
+        python datadog_search.py --regex tnl
+
+    Note: The search ignores case since most features are case insensitive.
+
+    Pre-requisites:
+    1. Install the client library:
+        pip install datadog-api-client
```

Comment:
The docs say to just run with `--help`, but this won't print out if the API client library isn't installed. Might be best to do a try-import (or just import from within the relying functions) to allow deferring that dependency until after the help message can be printed. Other options too.